nfv_sriov.md

```diff
@@ -0,0 +1,253 @@
+by adding it to the `configMaps` list.
+```yaml
+---
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneService
+metadata:
+  name: nova-custom-sriov
+spec:
+  label: dataplane-deployment-nova-custom-sriov
+  configMaps:
+    - cpu-pinning-nova
+    - sriov-nova
+  secrets:
+    - nova-cell1-compute-config
+  playbook: osp.edpm.nova
+```
+The custom service is named `nova-custom-sriov`. It cannot be named
+`nova` because `nova` is an immutable default service and will
+overwrite any custom service with the same name during reconciliation.
+
+After the `ConfigMap` and `OpenStackDataPlaneService` services above
+have been created (e.g. `oc create -f nova-custom-sriov.yaml`), update the
+`OpenStackDataPlaneNodeSet`
+[EDPM services list](https://openstack-k8s-operators.github.io/dataplane-operator/composable_services)
+to replace the `nova` service with `nova-custom-sriov`.
+
+```yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneNodeSet
+spec:
+  ...
+  roles:
+    edpm-compute:
+      ...
+      services:
+        - bootstrap
+        - download-cache
+        - configure-network
+        - validate-network
+        - install-os
+        - configure-os
+        - ssh-known-hosts
+        - run-os
+        - reboot-os
+        - install-certs
+        - libvirt
+        - ovn
+        - neutron-ovn
+        - nova-custom-sriov
+        - neutron-sriov
+        - neutron-metadata
+```
+The `bootstrap`, `configure-network` and `neutron-sriov` services
+should be added before the `libvirt` and `nova` (or `nova-custom-sriov`
+in this case) services. It configures EDPM nodes with SRIOV.
+
+When the `nova-custom-sriov` service Ansible job runs, it will copy
+overrides from the `ConfigMap`s onto the Nova hosts.
+
+### Create the OpenStackDataPlaneNodeSet
+
+Create the CR from your directory based on the example
+[dataplane_v1beta1_openstackdataplanenodeset_sriov](https://github.com/openstack-k8s-operators/dataplane-operator/tree/main/examples/sriov)
+with the changes described in the previous section.
+```
+oc kustomize --load-restrictor LoadRestrictionsNone dataplane-operator/examples/sriov > dataplane_cr.yaml
+```
+
+### Create a OpenStackDataPlaneDeployment
+
+Creating an `OpenStackDataPlaneDeployment` will trigger Ansible jobs
+to configure an EDPM. Which Ansible roles are run depends on the
+`services` list.
+
+Each `OpenStackDataPlaneDeployment` can have its own
+`servicesOverride` list which will redefine the list
+of services of an `OpenStackDataPlaneNodeSet` for a
+deployment.
+
+The example
+[dataplane_v1beta1_openstackdataplanedeployment_sriov](https://github.com/openstack-k8s-operators/dataplane-operator/tree/main/examples/sriov)
+
+Create the CR based on the example
+```
+oc kustomize --load-restrictor LoadRestrictionsNone dataplane-operator/examples/sriov > dataplane_cr.yaml
+```
+
+Custom `OpenStackDataPlaneService` called `nova-custom-sriov`
```

Comment:
I'm a little confused here.  I followed the links but its not clear to me what the nova-custom-sriov dataplane service definition should look like.

Reply:
never mind .. need to look up ..